app/controller/command/commands/export.py

```diff
@@ -0,0 +1,136 @@
+        self.gcp = gcp
+        self.bot = bot
+
+    def init_subparsers(self) -> _SubParsersAction:
+        """
+        Initialize subparsers for export command.
+
+        :meta private:
+        """
+        subparsers = self.parser.add_subparsers(dest="which")
+
+        # Parser for emails command
+        parser_view = subparsers.add_parser("emails")
+        parser_view.set_defaults(which="emails",
+                                 help="(Admin only)Export emails of all users")
+
+        return subparsers
+
+    def get_help(self, subcommand: str = None) -> str:
+        """Return command options for user events with Slack formatting."""
+
+        def get_subcommand_help(sc: str) -> str:
+            """Return the help message of a specific subcommand."""
+            message = f"\n*{sc.capitalize()}*\n"
+            message += self.subparser.choices[sc].format_help()
+            return message
+
+        if subcommand is None or subcommand not in self.subparser.choices:
+            res = f"\n*{self.command_name} commands:*```"
+            for argument in self.subparser.choices:
+                res += get_subcommand_help(argument)
+            return res + "```"
+        else:
+            res = "\n```"
+            res += get_subcommand_help(subcommand)
+            return res + "```"
+
+    def handle(self,
+               command: str,
+               user_id: str) -> ResponseTuple:
+        """Handle command by splitting into substrings and giving to parser."""
+        logging.debug("Handling ExportCommand")
+        command_arg = shlex.split(command)
+        args = None
+
+        try:
+            args = self.parser.parse_args(command_arg)
+        except SystemExit:
+            all_subcommands = list(self.subparser.choices.keys())
+            present_subcommands = [subcommand for subcommand in
+                                   all_subcommands
+                                   if subcommand in command_arg]
+            present_subcommand = None
+            if len(present_subcommands) == 1:
+                present_subcommand = present_subcommands[0]
+            return self.get_help(subcommand=present_subcommand), 200
+
+        if args.which == "emails":
+            try:
+                user_command = self.facade.retrieve(User, user_id)
+                if user_command.permissions_level == Permissions.admin:
+                    return self.export_helper()
+                else:
+                    return self.permission_error, 200
+            except LookupError:
+                return self.lookup_error, 200
+
+        else:
+            return self.get_help(), 200
+
+    def export_helper(self) -> ResponseTuple:
+        """
+        Export emails of all users as a string +
+        names of the users who do not have an email
+
+        """
+
+        params = [('permission_level', lvl) for lvl in ['admin', 'member']]
+        users = self.facade.query_or(User, params)
+
+        emails = ""
```

Comment:
`emails` needs a length check - if it exceeds this limit, rrom https://api.slack.com/docs/rate-limits#rate-limits__rtm-apis__posting-messages:

> Clients should limit messages sent to channels to 4000 characters

The email list should be separated appropriately (e.g. no chopping emails mid-word, so you'll want to check as you go)

Reply:
@bobheadxi I just tested it and looks like it can deliver up to 16000 characters! Do you think we are gonna exceed that limit?

Reply:
whoah that's generous! regardless, I would add a little something _just in case_ - at the very least, Rocket should say something like "whoah there are too many emails". I'll leave it up to you!

Reply:
I think another thing that may or may not be worth looking into is just uploading the CSV directly using the file upload system and linking the file in the reply, if we are afraid of going over the limit. Just a thought.

Reply:
<img width="1101" alt="Screen Shot 2020-11-06 at 16 58 41 PM" src="https://user-images.githubusercontent.com/14232149/98427281-642c5980-2051-11eb-8af9-d1c77b0b40e2.png">